webapp/tests/elster_client/test_elster_client.py

```diff
@@ -688,38 +688,58 @@ def test_none_for_date_keys_raises_value_error(self):
 
-        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True)):
+        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True, is_authenticated=True)):
             result = _generate_est_request_data(unknown_keys)
 
         self.assertEqual(unknown_keys, result['est_data'])
 
     def test_set_year_results_in_correct_year_in_result(self):
-        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True)):
+        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True, is_authenticated=True)):
             result = _generate_est_request_data({}, 2010)
 
         self.assertEqual(result['meta_data']['year'], 2010)
         self.assertEqual(result['meta_data']['is_digitally_signed'], True)
 
     def test_unset_year_results_in_2020_year_value(self):
-        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True)):
+        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True, is_authenticated=True)):
             result = _generate_est_request_data({})
 
         self.assertEqual(result['meta_data']['year'], 2020)
 
     def test_activated_user_results_in_correct_attribute_set_true(self):
-        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True)):
+        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True, is_authenticated=True)):
             result = _generate_est_request_data({})
 
         self.assertEqual(True, result['meta_data']['is_digitally_signed'])
 
     def test_inactive_user_results_in_correct_attribute_set_false(self):
-        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=False)):
+        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=False, is_authenticated=True)):
+            result = _generate_est_request_data({})
+
+        self.assertEqual(False, result['meta_data']['is_digitally_signed'])
+        
+    def test_not_authenticated_user_results_in_correct_attribute_set_false(self):
+        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=True, is_authenticated=False)):
             result = _generate_est_request_data({})
 
         self.assertEqual(False, result['meta_data']['is_digitally_signed'])
 
     def test_if_inactive_user_logout_user_is_called(self):
-        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=False)), \
+        with patch('app.elster_client.elster_client.current_user', MagicMock(is_active=False, is_authenticated=False)), \
+                patch('app.elster_client.elster_client.logout_user') as logout_fun:
+            _generate_est_request_data({})
+
+            logout_fun.assert_called_once()
+            
+    def test_if_not_authenticated_but_active_user_logout_user_is_called(self):
```

Comment:
For the unsuccessful cases I would add tests to ensure that nothing else happens (if we change the function accordingly)